core/src/main/java/nl/nn/adapterframework/filesystem/MsalClientAdapter.java

```diff
@@ -0,0 +1,222 @@
+			return translateResponse(response, session);
+		} catch (Exception e) {
+			log.error("An exception occurred whilst connecting with MSAL HTTPS call to [" + httpRequest.url().toString() + "]", e);
+			throw e;
+		} finally {
+			session.close();
+		}
+	}
+
+	private PipeLineSession prepareSession(HttpRequest httpRequest) {
+		PipeLineSession session = new PipeLineSession();
+		session.put(URL_SESSION_KEY, httpRequest.url().toString());
+		log.debug("Put request URL [{}] in session under key [{}]", ()->httpRequest.url().toString(), ()->URL_SESSION_KEY);
+
+		session.put(METHOD_SESSION_KEY, httpRequest.httpMethod().name());
+		log.debug("Put http method [{}] in session under key [{}]", ()->httpRequest.httpMethod().name(), ()->METHOD_SESSION_KEY);
+
+		session.put(REQUEST_HEADERS_SESSION_KEY, httpRequest.headers());
+		if(log.isDebugEnabled()) log.debug("Put http headers [{}] in session under key [{}]", ()->httpRequest.headers(), ()->REQUEST_HEADERS_SESSION_KEY);
+
+		return session;
+	}
+
+	@Override
+	protected HttpRequestBase getMethod(URI uri, Message message, ParameterValueList parameters, PipeLineSession session) throws SenderException {
+		HttpMethod httpMethod = EnumUtils.parse(HttpMethod.class, (String) session.get(METHOD_SESSION_KEY));
+		Map<String, String> headers = (Map<String, String>) session.get(REQUEST_HEADERS_SESSION_KEY);
+
+		if(uri == null) {
+			throw new SenderException("No URI to connect to!");
+		}
+
+		boolean queryParametersAppended = false;
+		StringBuffer relativePath = new StringBuffer(uri.getRawPath());
+		if(!StringUtils.isEmpty(uri.getQuery())) {
+			relativePath.append("?" + uri.getQuery());
+			queryParametersAppended = true;
+		}
+		try {
+			switch (httpMethod) {
+			case GET:
+				if(parameters != null) {
+					queryParametersAppended = appendParameters(queryParametersAppended, relativePath, parameters);
+					if(log.isDebugEnabled())
+						log.debug(getLogPrefix() + "path after appending of parameters [" + relativePath + "]");
+				}
+				HttpGet getMethod = new HttpGet(relativePath + (parameters == null ? message.asString() : ""));
+
+				if(log.isDebugEnabled())
+					log.debug(getLogPrefix() + "HttpSender constructed GET-method [" + getMethod.getURI().getQuery() + "]");
+				if(null != getFullContentType()) { // Manually set Content-Type header
+					getMethod.setHeader("Content-Type", getFullContentType().toString());
+				}
+
+				return appendHeaders(headers, getMethod);
+			case POST:
+				String messageString = message.asString();
+				if(parameters != null) {
+					StringBuffer msg = new StringBuffer(messageString);
+					appendParameters(true, msg, parameters);
+					if(StringUtils.isEmpty(messageString) && msg.length() > 1) {
+						messageString = msg.substring(1);
+					} else {
+						messageString = msg.toString();
+					}
+				}
+				HttpEntity entity = new ByteArrayEntity(messageString.getBytes(StreamUtil.DEFAULT_INPUT_STREAM_ENCODING), getFullContentType());
```

Comment:
Hier moet je een Byte Array uit de message halen

Reply:
Dit is een copy paste :D